sklearn/metrics/classification.py

```diff
@@ -1589,7 +1594,13 @@ def log_loss(y_true, y_pred, eps=1e-15, normalize=True, sample_weight=None):
     The logarithm used is the natural logarithm (base-e).
     """
     lb = LabelBinarizer()
-    T = lb.fit_transform(y_true)
+
+    
+
+    lb.fit(labels) if labels else lb.fit(y_true)
```

Comment:
This will raise an error if labels is a NumPy array, you should check `if labels is not None`